kafkaesk/app.py

```diff
@@ -246,16 +247,17 @@ async def __run(self) -> None:
                         error="UnhandledMessage",
                         group_id=self._subscription.group,
                     ).inc()
-
                 finally:
-                    context.span.finish()
                     context.close()
                     await self.emit("message", record=record)
+        except Exception as exc:
+            # Move the offset one position back so the failed message will be
+            # reprocessed.
+            tp = aiokafka.TopicPartition(record.topic, record.partition)
+            pos = await self._consumer.position(tp)
```

Comment:
Will `position` tell you the correct uncommitted value? I have a feeling this is going to look at what has been committed, not was is the working set.

I think you might want to look at this function: https://github.com/aio-libs/aiokafka/blob/master/aiokafka/consumer/subscription_state.py#L420

You might need to simply change the value that we would otherwise commit.

Another potential problem is that commit could happen between time that was spent processing the failure and when the failure propagates.

Really seems to me like the problem is still auto commit--it shouldn't record something as committable until after the `async for` yield value is done.

Reply:
thanks! ill change it and try the tests again, i wonder if now works just because auto commit is enabled.
i will try just overwriting the state value from the state, altought is really hacky and looks like something that will broke easily if aiokafka changes the api: `pos = self._consumer._subscription._assigned_state(tp).position`, so maybe after this we can also start thinking how to do it without auto commiting enabled.

Reply:
yah, it's really hacky :(

I think manual commit still might be better at this point.

Reply:
yeah, i dont like how it looks, i will see how difficult/easy is to disable auto commit and do it manually!